ast/TreeEquality.cc

```diff
@@ -10,18 +10,22 @@ namespace sorbet::ast {
 
 namespace {
 
-bool structurallyEqual(const core::GlobalState &gs, const void *avoid, const ExpressionPtr &tree,
-                       const ExpressionPtr &other, const core::FileRef file, bool root = false);
-
-bool structurallyEqualSpan(const core::GlobalState &gs, const void *avoid, absl::Span<const ExpressionPtr> a,
-                           absl::Span<const ExpressionPtr> b, const core::FileRef file) {
-    return absl::c_equal(a, b, [&](const auto &a, const auto &b) { return structurallyEqual(gs, avoid, a, b, file); });
+template <typename Comparator>
+bool traverse(const core::GlobalState &gs, const void *avoid, const ExpressionPtr &tree, const ExpressionPtr &other,
+              const core::FileRef file, bool root = false);
+
+template <typename Comparator>
+bool traverseSpan(const core::GlobalState &gs, const void *avoid, absl::Span<const ExpressionPtr> a,
+                  absl::Span<const ExpressionPtr> b, const core::FileRef file) {
+    return absl::c_equal(a, b,
+                         [&](const auto &a, const auto &b) { return traverse<Comparator>(gs, avoid, a, b, file); });
```

Comment:
I think I prefer the original names here: `traverse` is a little bit too generic. Given that it would be parameterized on the comparator, I think it's fine to keep the old name.

Reply:
It is generic, but it's still producing a comparison as the result. Maybe `traverseAndCompare` / `traverseAndCompareSpan`?

Reply:
Or perhaps `compareTrees` / `compareTreeSpans`?